akka-cluster-typed/src/test/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionistSpec.scala

```diff
@@ -296,5 +296,81 @@ class ClusterReceptionistSpec extends WordSpec with Matchers {
+      val config = ConfigFactory.parseString(
+        """
+          # disable delta propagation so we can have repeatable concurrent writes
+          # without delta reaching between nodes already
+          akka.cluster.distributed-data.delta-crdt.enabled=false
+        """).withFallback(ClusterReceptionistSpec.config)
+      val testKit1 = ActorTestKit("ClusterReceptionistSpec-test-5", config)
+      val system1 = testKit1.system
+      val testKit2 = ActorTestKit(system1.name, system1.settings.config)
+      val system2 = testKit2.system
+
+      val TheKey = ServiceKey[AnyRef]("whatever")
+      try {
+        val clusterNode1 = Cluster(system1)
+        val clusterNode2 = Cluster(system2)
+        clusterNode1.manager ! Join(clusterNode1.selfMember.address)
+        clusterNode2.manager ! Join(clusterNode1.selfMember.address)
+
+        val regProbe1 = TestProbe[AnyRef]()(system1)
+        val regProbe2 = TestProbe[AnyRef]()(system2)
+
+        regProbe1.awaitAssert(clusterNode1.state.members.count(_.status == MemberStatus.Up) == 2)
+
+        // one actor on each node up front
+        val actor1 = testKit1.spawn(Behaviors.receive[AnyRef] {
+          case (ctx, "stop") ⇒
+            ctx.log.info("Stopping")
+            Behaviors.stopped
+          case _ ⇒ Behaviors.same
+        }, "actor1")
+        val actor2 = testKit2.spawn(Behaviors.empty[AnyRef], "actor2")
+
+        system1.receptionist ! Register(TheKey, actor1)
+        system1.receptionist ! Subscribe(TheKey, regProbe1.ref)
+        regProbe1.awaitAssert(
+          regProbe1.expectMessage(Listing(TheKey, Set(actor1))),
+          5.seconds
+        )
+
+        system2.receptionist ! Subscribe(TheKey, regProbe2.ref)
+        regProbe2.fishForMessage(10.seconds) {
+          case TheKey.Listing(actors) if actors.nonEmpty ⇒
+            println(actors)
+            FishingOutcomes.complete
+          case _ ⇒ FishingOutcomes.continue
+        }
+        system1.log.info("Saw actor on both nodes")
+
+        // TheKey -> Set(actor1) seen by both nodes, now,
+        // remove on node1 and add on node2 (hopefully) concurrently
+        system2.receptionist ! Register(TheKey, actor2, regProbe2.ref)
+        actor1 ! "stop"
+        regProbe2.expectMessage(Registered(TheKey, actor2))
+        system2.log.info("actor2 registered")
+
+        // we should now, eventually, see the removal on both nodes
+        regProbe1.fishForMessage(10.seconds) {
+          case TheKey.Listing(actors) if actors.size == 1 ⇒
+            FishingOutcomes.complete
+          case _ ⇒
+            FishingOutcomes.continue
+        }
+        regProbe2.fishForMessage(10.seconds) {
+          case TheKey.Listing(actors) if actors.size == 1 ⇒
+            FishingOutcomes.complete
+          case _ ⇒
+            FishingOutcomes.continue
+        }
+
+      } finally {
+        testKit1.shutdownTestKit()
+        if (!system1.whenTerminated.isCompleted) testKit2.shutdownTestKit()
```

Comment:
why the `if`?

Reply:
copy-pasta from test above, I can remove it